lib/parsers/gjs-gts-parser.js

```diff
@@ -533,12 +538,88 @@ module.exports = {
     let jsCode = code;
     const info = transformForLint(code);
     jsCode = info.output;
+    jsCode = jsCode.replaceAll(/\.gts(["'])/g, '.ts$1 ');
 
     const isTypescript = options.filePath.endsWith('.gts');
 
     let result = null;
+
+    const projectFile = resolve(options.tsconfigRootDir, options.project);
+
+    const config = ts.getParsedCommandLineOfConfigFile(
+      projectFile,
+      {},
+      {
+        ...ts.sys,
+        readDirectory(dir) {
+          const results = ts.sys.readDirectory(dir);
+          return results.map((f) => f.replace(/\.gts$/, '.ts'));
+        },
+        onUnRecoverableConfigFileDiagnostic(diagnostic) {
+          let { messageText } = diagnostic;
+          if (typeof messageText !== 'string') {
+            messageText = messageText.messageText;
+          }
+
+          throw new Error(messageText);
+        },
+      }
+    );
+
+    const host = ts.createCompilerHost(config.options);
+    host.fileExists = function (fileName) {
+      try {
+        return fs.existsSync(fileName.replace(/\.ts$/, '.gts'));
+      } catch {
+        return fs.existsSync(fileName);
+      }
+    };
+
+    const readDirectory = host.readDirectory;
+    host.readDirectory = function(dir) {
+      const results = readDirectory.call(this, dir);
+      return results.map((f) => f.replace(/\.gts$/, '.ts'));
+    };
+    host.readFile = function(fileName) {
+      if (fileName === options.filePath.replace(/\\/g, '/').replace(/\.gts$/, '.ts')) {
+        return jsCode;
+      }
+      let content = fs.readFileSync(fileName).toString();
+      if (fileName.endsWith('.gts')) {
+        content = processor.process(content, fileName);
+      }
+      content = content.replaceAll(/\.gts(["'])/g, '.ts$1 ');
+      return content;
+    };
+
+    // host.resolveModuleNameLiterals = function(
+    //   moduleLiterals,
+    //   containingFile,
+    //   redirectedReference,
+    //   options
+    // ) {
+    //   return moduleLiterals.map((literal) => {
+    //     return ts.resolveModuleName(
+    //       literal.text,
+    //       containingFile,
+    //       options,
+    //       host,
+    //       {},
+    //       redirectedReference
+    //     );
+    //   });
+    // };
+
+    const program = ts.createProgram({
```

Comment:
why not just use glint at this point?

Reply:
I'm not sure if running `createProgram` every time `parseForEslint` is called is performant?
Should we maybe save and reuse program? Also probably need to do `createWatchProgram` if in IDE context?